plugin-server/src/utils/db/db.ts

```diff
@@ -497,27 +497,41 @@ export class DB {
         update: Partial<Person>,
         client?: PoolClient
     ): Promise<Person | ProducerRecord[]> {
-        const updatedPerson: Person = { ...person, ...update }
-        const values = [...Object.values(unparsePersonPartial(update)), person.id]
+        const updateValues = Object.values(unparsePersonPartial(update))
 
-        const queryString = `UPDATE posthog_person SET ${Object.keys(update).map(
-            (field, index) => `"${sanitizeSqlIdentifier(field)}" = $${index + 1}`
-        )} WHERE id = $${Object.values(update).length + 1}`
+        // short circuit if there are no updates to be made
+        if (updateValues.length === 0) {
+            return client ? [] : person
+        }
```

Comment:
We have this check in the process-event side already & I'm not sure why, but it's bypassed when we're using kafka producer i.e. Clickhouse DB.
We always pass the full person's properties to this function from `updatePersonProperties` function at least.